spec/lib/ruby_smb/client_spec.rb

```diff
@@ -1279,10 +1279,13 @@
       end
 
       it 'calls the backing methods' do
-        expect(client).to receive(:negotiate_request)
+        expect(client).to receive(:negotiate_request).twice.and_call_original
         expect(client).to receive(:send_recv)
         expect(client).to receive(:negotiate_response)
-        expect(client).to receive(:parse_negotiate_response)
+        expect(client).to receive(:parse_negotiate_response).twice do
+          client.dialect = '0x02ff'
+          client.smb1 = false
+        end
         client.negotiate
       end
```

Comment:
It looks like this example now checks something else it was originally created for. I believe it was to make sure the backing methods were called (even if it is not that useful, I agree). Now, it is checking the logic when a wildcard is received, which is good, but I think this is what is done in the `with a wildcard revision number response` context below.

That would be great to have a specific set of examples that make sure it handles correctly response from old Windows (2.0.2 instead of wildcard) and it does not break the logic for other dialects 2.1 and 3.1.1, as you described. Note that I think the `with 0x0311 dialect` context below already validate the latter.

That said, this is how I would fix this example with the changes in `#negotiate`:
```ruby
      it 'calls the backing methods' do
        request_packet = double('Request packet')
        allow(client).to receive(:negotiate_request).and_return(request_packet)
        allow(request_packet).to receive(:packet_smb_version)
        expect(client).to receive(:negotiate_request)
        expect(client).to receive(:send_recv)
        expect(client).to receive(:negotiate_response)
        expect(client).to receive(:parse_negotiate_response)
        client.negotiate
      end
```